src/Algebra/Structures.agda

```diff
@@ -107,31 +107,73 @@ record IsMonoid (∙ : Op₂ A) (ε : A) : Set (a ⊔ ℓ) where
 
 
 record IsCommutativeMonoid (∙ : Op₂ A) (ε : A) : Set (a ⊔ ℓ) where
+  field
+    isMonoid : IsMonoid ∙ ε
+    comm     : Commutative ∙
+
+  open IsMonoid isMonoid public
+
+  isCommutativeSemigroup : IsCommutativeSemigroup ∙
+  isCommutativeSemigroup = record
+    { isSemigroup = isSemigroup
+    ; comm        = comm
+    }
+
+record IsCommutativeMonoidˡ (∙ : Op₂ A) (ε : A) : Set (a ⊔ ℓ) where
```

Comment:
I'm wondering if you really want these biased versions to be records. Since you mention that the only purpose of these is constructing instances of `IsCommutativeMonoid`, perhaps a function would be more appropriate.

Reply:
I'm definitely in favour of keeping them as records to maximise backwards compatibility.

Reply:
Even without caring about backwards compatibility, I think this way works best. If there were no record definition, these smart constructors would quickly get unreadable due to the number of arguments they take. They would also disrupt the usual layout of algebraic structure instances.

Reply:
:+1: